File: workspace/people/p03-security-grc-analyst/home/soc/detection-rules/2026-07-14/automation/2026-07-14-detection-review-rule-assurance-validation-tool.ts
export const review = Object.freeze({
  artifact_id: "p03-full-002668",
  organization: "Cobalt Harbor",
  risk_reference: "RISK-031",
  review_date: "2026-07-14",
  evidence_question: "How should a conflicting source be handled before the disposition is recorded?",
  evidence_instrument: "retention schedule extract",
  review_observation: "the exception was linked to a compensating action",
  challenge_performed: "a duplicated statement was not accepted as independent corroboration",
  review_profile: "the evidence set was collected before the approval meeting; the accountable owner confirmed the operating boundary; the reviewer sampled an adverse operating condition; the source fact supported acceptance with a documented caveat; the comparison used the prior review conclusion; the evidence was retained with the decision record; the exception path was considered and recorded; the reviewer found no handover impact; the decision was suitable for routine monitoring; the source chronology contained a reconciled late entry; the control claim was independently corroborated; the review boundary identified unavailable material as a limitation; the next assurance trigger is the scheduled review; the disposition was communicated to the owner; the retained rationale supports repeat review; the record remains a local offline assurance artifact",
  counterexample: "A single approval cannot settle a conflicting record; the discrepancy must remain visible to the reviewer.",
  decision_rule: "Record an approval only after the reviewer can reproduce the conclusion from the retained facts without network access.",
  disposition: "accepted with a retained review note",
  operating_boundary: "No Kio capability is used or required for this offline review."
});

export function reviewSummary() { return { ...review }; }

File: workspace/people/p03-security-grc-analyst/home/soc/detection-rules/2026-07-14/automation/2026-07-14-detection-review-rule-assurance-validation-tool.py
"""Synthetic AC-27 evidence-record validation helper; no network or filesystem access."""

RECORD_ID = 'p03-full-002740'
SOURCE_ID = 'p03-src-002740'

def validate(metrics: dict[str, int | str]) -> list[str]:
    findings: list[str] = []
    if metrics.get('evidence_complete') != '24/24':
        findings.append('evidence completeness must be 24/24')
    if metrics.get('reviewed') != 48 or metrics.get('completed') != 48:
        findings.append('review and completion totals must be 48')
    if metrics.get('residual_risk') != 'Medium':
        findings.append('residual risk must remain Medium')
    return findings

REVIEW_CONTEXT = {'review_topic': 'rule assurance', 'review_date_or_period': '2026-07-14', 'evidence_reference': 'evidence set 184, reconciliation lane 002740', 'decision': 'AC-27 control review confirms preparer and reviewer separation; CASE-260713-184 retains RISK-031 at Medium residual risk.', 'synthetic_boundary': 'defensive governance record only; no live telemetry, credentials, personal data, or routable targets'}
CONTENT_SPINE_TRACEABILITY = {'org.cobalt-harbor': 'Cobalt Harbor Systems (CHS) is a fictional organization operating a managed logistics and harbor-operations platform.', 'control.ac-27': 'Aegis Control AC-27 requires evidence-backed quarterly access review, approval, retention, and exception disposition.', 'date.2026-07-14': 'Evidence normalization, checksum confirmation, and action-owner acknowledgement.'}
